### PR TITLE
Dev/ccelik/dedicated host swagger

### PR DESCRIPTION
**Pull request by celikcigdem**
Added new APIs for DedicatedHostGroups and DedicatedHosts

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [X] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [X] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by anuchandy** [comment]
@sarangan12 @sergey-shandar linter emitted below error:

```
R2020 - RequiredPropertiesMissingInResourceModel
Model definition 'DedicatedHostGroupUpdate' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly. 
at specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json:5491
```

This is the PATCH request payload and is not used as response model.

However this type derive `UpdateResource` type which is marked as  "x-ms-azure-resource", I don't see a need to mark it so though. Do you think linter emit this error due to this annotation? or anything else causing this error to trigger?

**Review by sarangan12 — CHANGES_REQUESTED** [review]
The new file is not referenced in the readme file. Please update it

**Comment by celikcigdem** [comment]
> The new file is not referenced in the readme file. Please update it

@sarangan12, @anuchandy, I resolved your comments, the new iteration is available. Could you please review it?

**Comment by celikcigdem** [comment]
@hyonholee , @huangpf could you please review this PR?

**Review by sarangan12 — APPROVED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json [hunk 1]
```diff
@@ -4928,6 +5449,267 @@
...
+      },
+      "description": "Properties of the dedicated host."
+    },
+    "DedicatedHost": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DedicatedHostProperties"
+        },
+        "sku": {
```
Comment (adewaleo) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: Should this be sku or something else?

Also this property / its definition was not defined

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json [hunk 2]
```diff
@@ -4928,6 +5449,267 @@
...
+          "$ref": "#/definitions/Sku",
```
Comment (adewaleo) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: @celikcigdem @sarangan12, should this be #/definitions/sku or something else?

This definition does not exist

Comment (adewaleo) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: Correction: this Sku exists. Its comment says it describes a virtual machine scale set sku. I will use it for now, but I would have imagined that a new type DedicatedHostSkuTypes would be defined for this property.

**Pull request merged**